src/compiler/msvc.rs

```diff
@@ -340,13 +350,15 @@ pub fn preprocess<T>(creator: &T,
 
     let parsed_args = parsed_args.clone();
     let includes_prefix = includes_prefix.to_string();
-    let cwd = cwd.to_string();
+    let cwd = cwd.to_owned();
 
     Box::new(run_input_output(cmd, None).and_then(move |output| {
         let parsed_args = &parsed_args;
         if let (Some(ref objfile), &Some(ref depfile)) = (parsed_args.outputs.get("obj"), &parsed_args.depfile) {
-            let mut f = File::create(Path::new(&cwd).join(depfile))?;
-            write!(f, "{}: {} ", objfile, parsed_args.input)?;
+            let mut f = File::create(cwd.join(depfile))?;
+
+            // TODO: these calls to `display` are lossy ...
```

Comment:
Oh looks like `WideCharToMultiByte` may actually [be supported](https://docs.rs/local-encoding/0.2.0/i686-pc-windows-gnu/local_encoding/windows/fn.wide_char_to_multi_byte.html), I'll play around with calling that (never tried it before!)